src/wazuh_db/wdb_agent.c

```diff
@@ -914,61 +753,36 @@ int wdb_delete_agent_belongs(int id_agent) {
 }
 
 int wdb_update_groups(const char *dirname) {
-    int result =  0;
+    int result = 0;
     int i = 0;
     int n = 0;
     char **array = NULL;
-    char *json_string = NULL;
     cJSON *elem = NULL;
     cJSON *name = NULL;
     cJSON *root = NULL;
     char wdbquery[OS_BUFFER_SIZE] = "";
-    char wdboutput[OS_MAXSTR] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
      
     sqlite3_snprintf(sizeof(wdbquery), wdbquery,"%s", global_db_queries[SQL_SELECT_GROUPS]);
-    result = wdbc_query_ex( &wdb_sock_agent, wdbquery, wdboutput, sizeof(wdboutput));
- 
-    switch (result){
-        case OS_SUCCESS:
-            break;
-        case OS_INVALID:
-            mdebug1("GLobal DB Error in the response from socket");
-            mdebug2("Global DB SQL query: %s", wdbquery);
-            return OS_INVALID;
-        default:
-            mdebug1("GLobal DB Cannot execute SQL query; err database %s/%s.db", WDB2_DIR, WDB2_GLOB_NAME);
-            mdebug2("Global DB SQL query: %s", wdbquery);
-            return OS_INVALID;
-    }
-
-    if(json_string = wstr_chr(wdboutput, ' '), json_string ){
-        *json_string='\0';
-        json_string++;
+    root = wdbc_query_parse_json(&wdb_sock_agent, wdbquery, wdboutput, sizeof(wdboutput));
 
-    } else{
-        mdebug1("SQLite result has no space. Query: %s",wdbquery);
-        os_free(array);
+    if (!root) {
+        merror("Error querying Wazuh DB to update agent group.");
         return OS_INVALID;
     }
 
-    if(strcmp(wdboutput,"ok") == 0 && strcmp(json_string,"[]") != 0 && strcmp(json_string,"[{}]") != 0){
-        root = cJSON_Parse(json_string);
-        n = cJSON_GetArraySize(root);
-        os_calloc(n+1, sizeof(char *),array);        
-        
-        for (i = 0; i < n; i++) {
-            elem = cJSON_GetArrayItem(root, i);
-            name = cJSON_GetObjectItem(elem, "name");
-            os_strdup(name->valuestring,array[i]);
-            }
-        array[i] = NULL;
+    n = cJSON_GetArraySize(root);
+    os_calloc(n+1, sizeof(char *),array);        
 
-    } else{
-        mdebug1("SQLite Query failed: %s", wdbquery);
-        os_free(array);
-        return OS_INVALID;
+    for (i = 0; i < n; i++) {
+        elem = cJSON_GetArrayItem(root, i);
+        name = cJSON_GetObjectItem(elem, "name");
+        os_strdup(name->valuestring,array[i]);
```

Comment:
Can cJSON_GetObjectItem return NULL value? Should we check it?

Reply:
The new implementation considers these cases according to the documentation
[cJSON](https://github.com/DaveGamble/cJSON)